lib/std/build.zig

```diff
@@ -1730,6 +1741,7 @@ pub const LibExeObjStep = struct {
         self.link_objects.append(LinkObject{ .StaticPath = self.builder.dupe(path) }) catch unreachable;
     }
 
+    /// Add a dependency on an object file.
```

Comment:
This is misleading, since a build dependency is a separate concept from adding an object into a compilation.